File: p/1/1.pas
procedure one;
begin
  var (n, p) := (ReadInteger, 1);
  var a: integer;
  
  while n <> 0 do
  begin
    a := n mod 10;
    p *= a;
    n := n div 10;
  end;
  
  Println(p);
end;


procedure two;
begin
  var (x, k, n) := (1, 0, readinteger);
  
  while x <= n do
  begin
    if n mod x = 0 then
      k += 1;
    x += 1;
  end;
  Println(k);
end;


procedure three;
begin
  var (n, x, s) := (readinteger, 1, 0);
  while x <= n do
  begin
    if n mod x = 0 then
      s += x;
    x += 1;
  end;
  if s = n then
    Println('ДА')
  else
    Println('НЕТ');
end;


procedure four;
begin
  var (i, k, n) := (1, 0, ReadInteger('Введите n: '));
  var x, c, y: integer;
  
  while i <= n do
  begin
    (x, c) := (1, 0);
    y := ReadInteger('Введите y: ');
    
    while x <= y do
    begin
      if y mod x = 0 then
        c += 1;
      x += 1;
    end;
    
    if c = 2 then
      k += 1;
    i += 1;
  end;
  
  Println(k);
end;


function is_perfect_num(num: integer): boolean;
begin
  var total := 1;
  
  for var i := 2 to round(sqrt(num))+1 do
  begin
    if num mod i = 0 then
      total += i + (num div i);
  end;
  Result := total = num;
end;

procedure five;
begin
  var n := ReadInteger('Введите количество элементов массива');
  var arr := ReadArrInteger(n);
  
  Println(arr.Where(x -> is_perfect_num(x)).Product);
end;


begin
  five;
end.

File: p/1/1.py
# Дано натуральное число n. Найдите произведение всех его цифр.
def one():
    n, p = int(input()), 1
    while n != 0:
        a = n % 10
        p *= a
        n = n // 10
    print(p)


# Дано натуральное число n. Найдите количество его положительных делителей.
def two():
    n = int(input())
    x, k = 1, 0
    
    while x <= n:
        if n % x == 0:
            k += 1
        x += 1
    print(k)


# Дано натуральное число n. Определите, является ли оно совершенным.
def three():
    n, x, s = int(input()), 1, 0
    while x <= n:
        if n % x == 0:
            s += x
        x += 1
    print('ДА' if s==n else 'НЕТ')


# Дано число n. Затем вводится n натуральных чисел (по одному на строку).
# Определите, сколько из этих чисел являются простыми.
def four():
    i, k, n = 1, 0, int(input('Введите n: '))
    
    while i <= n:
        x, c = 1, 0
        y = int(input('Введите y: '))
        
        while x <= y:
            if y % x == 0:
                c += 1
            x += 1
            
        if c == 2:
            k += 1
        i += 1
    print(k)


def is_perfect_num(num: int) -> bool:
    return sum(i + num//i for i in range(2, int(num**0.5) + 1) if num % i == 0) + 1 == num


def five():
    from math import prod
    print(prod([i for i in list(map(int, input().split())) if is_perfect_num(i)]))


if __name__ == '__main__':
    five()